File: schrottverwaltung/src/main/java/de/fhdortmund/schrottverwaltung/immobilie/entity/ImmoStatus.java
package de.fhdortmund.schrottverwaltung.immobilie.entity;

import de.fhdortmund.schrottverwaltung.immobilie.ImmoStatusEnum;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "immobilien_status")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ImmoStatus {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "immobilien_status_id")
    private Long id;

    @Enumerated(EnumType.STRING)
    private ImmoStatusEnum status;

    @Column(name = "beschreibung")
    private String beschreibung;

    @ManyToOne
    @JoinColumn(name = "immobilie_id") // <- Referenz zur Immobilie
    private Immobilie immobilie;
}

File: schrottverwaltung/src/main/kotlin/de/fhdortmund/schrottverwaltung/immoblilie/ImmoStatus.kt
package de.fhdortmund.schrottverwaltung.immoblilie

import jakarta.persistence.*

@Entity
@Table(name = "immobilien_status")
data class ImmoStatus(
    @Enumerated(EnumType.STRING)
    var status: ImmoStatusEnum,
    @Column(name = "beschreibung")
    var beschreibung: String,
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "immobilien_status_id")
    var id: Long? = null
)
